enterprise/core-edge/src/main/java/org/neo4j/coreedge/catchup/storecopy/edge/StoreCopyClient.java

```diff
@@ -58,4 +60,19 @@ public long copyStoreFiles( CoreMember from, StoreFileStreams storeFileStreams )
             coreClient.removeStoreFileStreamingCompleteListener( fileStreamingCompleteListener );
         }
     }
+
+    public StoreId fetchStoreId( CoreMember from ) throws StoreIdDownloadFailedException
+    {
+        CompletableFuture<StoreId> storeIdCompletableFuture = new CompletableFuture<>();
+        coreClient.setStoreIdConsumer( storeIdCompletableFuture::complete );
+        coreClient.requestStoreId( from );
+        try
+        {
+            return storeIdCompletableFuture.get();
```

Comment:
Yes I am sure since everything in around store copy has unbounded waits, so I am not gonna change that in this PR. I might consider to revise all the unbound waits related to store copy and issue a PR for that.